staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -81,7 +82,9 @@ func (p *Preferences) AddFlags(flags *pflag.FlagSet) {
 // Apply firstly applies the aliases in the preferences file and secondly overrides
 // the default values of flags.
 func (p *Preferences) Apply(rootCmd *cobra.Command, args []string, errOut io.Writer) ([]string, error) {
-	if len(args) <= 1 {
+	if len(args) <= 1 || (len(args) == 2 && (args[1] == "--help" || args[1] == "-h")) {
```

Comment:
This is error prone and not quite sufficient. If I do `kubectl --kuberc mykuberc -h` the above won't work. Or in general any mixture of `-h` with global flags. It'd be better to iterate over args and verify if `-h` or `--help` is available, and only when no command is present, because that's entirely different case as well.